### DISCUSSION TITLE
Unable to get data from AdGuard Home

### DISCUSSION THREAD

**Discussion by s3gfaultx** [linuxserver/Heimdall#1004] [App issues]
No matter how I have things configured, I get the following error when clicking test:
- Something went wrong... Code: 403

I've verified username/password and tried all variations of the URL.

AdGuard Home version v0.107.6
Heimdall version 2.4.11

**Comment by Tropaion** [comment]
I have the same error

**Comment by Curiosity87** [comment]
Same here. Something went wrong... Code: 403.

Check username/password and tried all variations of the URL.

**Comment by lkoning** [comment]
I have the same problem. Is there anything to find on how to configure this?

**Comment by lggomez** [comment]
For me it was working on a fresh install but after configuring adding a few other apps it stopped working, even with the valid credentials; Test gives a modal with the `General error connecting with API` message

Is there a way to get logs on this to see if I can figure something else up?

**Comment by KodeStar** [comment]
Try setting debug to true in the .env file

**Comment by M0UL** [comment]
from what i can tel it is trying to connect to http://username:password@ip:port/control/stats. this does not work
anybody got this to work?

**Comment by rajmondburgaj** [comment]
Same issue here, does not work!

**Comment by ve6rah** [comment]
I was having the same problem, if you have any special characters in your username or password, you'll need to URL encode them. Adguard is passing all this as a URL, and you can't have special characters in those.

**Reply by rajmondburgaj** [reply]
Thanks for the advice, but still did not work after encoding the username and password.

**Reply by gladiatormax** [reply]
Same here, just did a fresh installation off Adguard :-(

**Reply by ajax1337** [reply]
Thanks it worked for me : used https://www.urlencoder.org for encoding

**Reply by ZoltrixGFC** [reply]
Wow, this actually works. I've been trying to fix it for ages. As well as converting the password, I found out you also needed to use the internal docker port for it to connect. This only applies if you are using a reverse proxy on the same docker host, etc.
